xla/service/llvm_ir/llvm_util.cc

```diff
@@ -377,8 +378,11 @@ llvm::AllocaInst* EmitAllocaAtFunctionEntryWithCount(llvm::Type* type,
   llvm::Function* function = b->GetInsertBlock()->getParent();
   b->SetInsertPoint(&function->getEntryBlock(),
                     function->getEntryBlock().getFirstInsertionPt());
+  llvm::Module* module = b->GetInsertBlock()->getModule();
+  // Explicitly set private addrspace for SPIR target.
+  int addrspace = llvm::Triple(module->getTargetTriple()).isSPIR() ? 5 : 0;
```

Comment:
Could you please document (in the comment) what address space 5 means for SPIR? It's specific to Intel Accelerators, right? It would be great if you could add the link to the specification, e.g., this 
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/docs/SPIRVRepresentationInLLVM.rst#address-spaces

Reply:
This change is not related to addrspace representation in spir. Just as the title mentioned, we need explicitly set private addrspace(5) instead of generic addrspace(0, by default) for SPIR target, so I made this change.

More words about addrspace representations in spir. spir backend uses different addrspace representations [link](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/0aab1249acbfb8d206c24988f0974638c290e931/lib/SPIRV/SPIRVInternal.h#L181) with nvptx backend [link](https://llvm.org/docs/NVPTXUsage.html#address-spaces). To make xla codegen simple(avoiding changing addrspace based on device backend everywhere), I plan to use nvptx addrspace representations for spirv backend, and do a mapping from nvptx addrspace to spirv in SPIRV-LLVM-Translator, which only needs several lines changes. But this is not related to this PR, I will submit another PR to cover this.

Reply:
Got it. Thank you for the explanation!